packages/core/core/src/requests/BundleGraphRequest.js

```diff
@@ -202,52 +253,16 @@ class BundlerRunner {
     let plugin = await this.config.getBundler();
     let {plugin: bundler, name, resolveFrom} = plugin;
 
-    let {cacheKey, bundlerHash} = await this.getHashes(graph);
-
-    // Check if the cacheKey matches the one already stored in the graph.
-    // This can save time deserializing from cache if the graph is already in memory.
-    // This will only happen in watch mode. In this case, serialization will occur once
-    // when sending the bundle graph to workers, and again on shutdown when writing to cache.
-    let previousResult = await this.api.getPreviousResult(cacheKey);
-    if (previousResult != null) {
-      // No need to call api.storeResult here because it's already the request result.
-      return previousResult;
-    }
-
-    // Otherwise, check the cache in case the cache key has already been written to disk.
-    if (!this.options.shouldDisableCache) {
-      let cached = await this.options.cache.getBuffer(cacheKey);
-      if (cached != null) {
-        // Deserialize, and store the original buffer in an in memory cache so we avoid
-        // re-serializing it when sending to workers, and in build mode, when writing to cache on shutdown.
-        let graph = deserializeToCache(cached);
-        this.api.storeResult(
-          {
-            bundleGraph: graph,
-            changedAssets: new Map(),
-          },
-          cacheKey,
-        );
-        return graph;
-      }
-    }
-
```

Comment:
These cases were never hit at all in the tests, and should be very rare. I think removing the traversal to compute the hashes in the common case outweighs the potential benefits for zero change builds.